hibernate-core/src/main/java/org/hibernate/query/criteria/internal/path/AbstractFromImpl.java

```diff
@@ -81,12 +81,7 @@ protected boolean canBeDereferenced() {
 	@Override
 	public void prepareAlias(RenderingContext renderingContext) {
 		if ( getAlias() == null ) {
```

Comment:
I'm not sure this will generate the correct HQL. Maybe this should be the following
```suggestion
		if ( super.getAlias() == null ) {
```

and you should revert the other changes in this file

Reply:
I think it should be fine. I removed the `getAlias()` overloading method in this class (to ensure new alias is generated rather than reusing the correlated parent's alias), so both `getAlias()` and 'setAlias()` would be from the parent class.

But as you suggested, we need to revert back partially to the previous implementation for performance reason (only create new alias when the correlated root is of LEFT JOIN), then I need to reconsider the issue.